commit b24da0142bd3babba1935866b266845ed2ece4c6
Author: Ola x Nilsson <olani@axis.com>
Date:   Fri Mar 17 15:30:46 2023 +0100

    gnu: Handle timeval.tv_usec for glibc 64-bit time_t
    
    For 64 bit time on 32 bit linux glibc timeval.tv_usec is actually
    __suseconds64_t (64 bits) while suseconds_t is still 32 bits.

diff --git a/src/unix/mod.rs b/src/unix/mod.rs
index a380bf1ef8..42a293e9b0 100644
--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -62,7 +62,12 @@ s! {
 
     pub struct timeval {
         pub tv_sec: time_t,
+        #[cfg(not(gnu_time64_abi))]
         pub tv_usec: suseconds_t,
+        // For 64 bit time on 32 bit linux glibc, suseconds_t is still
+        // a 32 bit type.  Using suseconds_t here will break the tests, use i64 instead
+        #[cfg(gnu_time64_abi)]
+        pub tv_usec: i64
     }
 
     // linux x32 compatibility
